server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -283,6 +283,11 @@ public void createSegmentTable(final String tableName)
                 "CREATE INDEX idx_%1$s_datasource_used_end ON %1$s(dataSource, used, %2$send%2$s)",
                 tableName,
                 getQuoteString()
+            ),
+            StringUtils.format(
+                "CREATE INDEX idx_%1$s_datasource_used_start_end ON %1$s(dataSource, used, start, %2$send%2$s)",
+                tableName,
+                getQuoteString()
```

Comment:
Since this index contains all fields in the 2nd index above, can we delete the index above?

Reply:
We did that in production 2 weeks back on the same thesis, but quickly ingestion lag started going up and we had to recreate the index with (datasource, used, end) again to fix it. There is some dependency which we did not investigate further.

Reply:
@FrankChen021 - since the new index has "end" as the last column, may be queries that rely only on (datasource, used, end) fields got slow enough to induce ingestion lag in our environment. Since the symptoms were evident after deleting the index with (datasource, used, end) columns, I feel we should have both indexes. wdyt ?

Reply:
Sorry for the late reply. How about put the ‘start’ field after ‘end’ field so that it’s the prefix of (datasource, used, end) query? I think you could try to explain queries to see how the index is used.

Reply:
@FrankChen021 - Thanks for your recommendation, we tested the change of putting start at the end and deployed it in lab, stag and prod. All the envs are working fine without issues. Please take a look at the change now :)